CONTRIBUTING.md

```diff
@@ -0,0 +1,151 @@
+
+# How to contribute to the Azure SDK for Javascript
+
+There are many ways that you can contribute to the Azure SDK for JavaScript project:
+
+- Submit a bug
+- Submit a code fix for a bug
+- Submit additions or modifications to the documentation
+- Submit a feature request
+
+All code submissions will be reviewed and tested by the team, and those that meet a high bar for both quality and design/roadmap appropriateness will be merged into the source. Be sure to follow the existing file/folder structure when adding new boards or sensors.
+
+If you encounter any bugs with the library please file an issue in the [Issues](https://github.com/Azure/azure-sdk-for-js/issues) section of the project.
+
+## Things to keep in mind when contributing
+
+Some guidance for when you make a contribution:
+
+- Add/update unit tests and code as required by your change
+- Make sure you run all the unit tests on the affected platform(s)/languages. If the change is in common code, generally running on one platform would be acceptable.
+- Run end-to-end tests or simple sample code to make sure the lib works in an end-to-end scenario.
+
+## Big contributions
+
+If your contribution is significantly big it is better to first check with the project developers in order to make sure the change aligns with the long term plans. This can be done simply by submitting a question via the GitHub Issues section.
+
+Want to get started hacking on the code? Great! Keep reading. This contributing guide helps you to start working with and contributing to the Azure Typescript/Javascript SDK Generator.
+
+### General Autorest generation design
+
+_TBA_
+
+### Typescript/Javascript SDK generator Design
+
+The generated Typescript/Javascript SDK has the following structure:
+
+1. Client (Handled by [clientFileGenerator.ts](./src/generators/clientFileGenerator.ts))
+2. Client Context (Handled by [clientContextFileGenerator.ts](./src/generators/clientContextFileGenerator.ts))
+3. Operation Interfaces (Handled by [operationInterfaceGenerator.ts](./src/generators/operationInterfaceGenerator.ts))
+4. Operations (Handled by [operationGenerator.ts](./src/generators/operationGenerator.ts))
+5. Models (Handled by [modelsGenerator.ts](./src/generators/modelsGenerator.ts))
+6. Parameters (Handled by [parametersGenerator.ts](./src/generators/parametersGenerator.ts))
+7. Index File (Handled by [indexGenerator.ts](./src/generators/indexGenerator.ts))
+8. Metadata (Handled [here](./src/generators/static))
+
+The entry point for this design is at [main.ts](./src/main.ts) which invokes the [typescriptGenerator.ts](./src/typescriptGenerator.ts) and starts the entire generation process. You could find all the options that could be sent to the generator at [autorestOptions.ts](./src/utils/autorestOptions.ts)
+
+On a high level, the entire SDK generation process would be:
+
+Swagger Input -> Autorest Core & Modeler -> CodeModel -> TS/JS SDK Generator -> Generated SDK
```

Comment:
this could be done better in ascii art since code model and generated SDK are data and not part of the pipeline

Reply:
We can do this later